src/hotspot/share/runtime/jniHandles.cpp

```diff
@@ -384,10 +384,15 @@ void JNIHandleBlock::release_block(JNIHandleBlock* block, JavaThread* thread) {
       block->_next = freelist;
     }
     block = NULL;
-  }
-  if (block != NULL) {
-    Atomic::dec(&_blocks_allocated);
-    delete block;
+  } else {
+    DEBUG_ONLY(block->set_pop_frame_link(NULL));
```

Comment:
Can this also be an assert that the pop_frame_link is null ?

Reply:
This is a first block which can have non-NULL set_pop_frame_link. So I set it to NULL so assert in the while loop doesn't fail.